site/cypress/e2e/sidebar.spec.js

```diff
@@ -0,0 +1,83 @@
+import { getCurrentSemester } from '../support/utils.js';
+
+const currentSemester = `${getCurrentSemester()}`;
+
+Cypress.Commands.add('sidebarContains', (title, extension) => {
+    cy.get('aside ul li').contains(title).should('have.attr', 'href').and('contain', extension);
+    if (!title.includes('Logout')) {
+        cy.get('aside ul li').contains(title).click({ force: true });
+        if (title.includes('Autograding')) {
+            cy.get('#main > .content > h1').should('contain', 'Job Statistics');
+        }
+        else if (title.includes('Discussion Forum')) {
+            cy.get('#main > .content').should('contain', 'Create Thread');
+        }
+        else if (title.includes('Collapse Sidebar')) {
+            return;
+        }
+        else {
+            cy.get('#main > .content').should('contain', title);
+        }
+    }
+});
+
+Cypress.Commands.add('baseSidebar', (user) => {
+    cy.sidebarContains('My Courses', '/home');
+    cy.sidebarContains('My Profile', '/user_profile');
+    cy.sidebarContains('Authentication Tokens', '/authentication_tokens');
+    cy.sidebarContains('Calendar', '/calendar');
+    if (user.includes('instructor')) {
+        cy.sidebarContains('Docker', '/admin/docker');
+        cy.sidebarContains('New Course', '/home/courses/new');
+        cy.sidebarContains('Autograding Status', '/autograding_status');
+    }
+    cy.sidebarContains('Collapse Sidebar', 'javascript: toggleSidebar();');
+    cy.sidebarContains('Logout', '/authentication/logout');
+});
+
+Cypress.Commands.add('testCourseSidebar', (user, course) => {
+    cy.login(user);
+    cy.visit([course]);
+    cy.sidebarContains('Gradeables', `/courses/${currentSemester}/sample`);
+    cy.sidebarContains('Notifications', `/courses/${currentSemester}/sample/notifications`);
+    cy.sidebarContains('Office Hours Queue', `/courses/${currentSemester}/sample/office_hours_queue`);
+    cy.sidebarContains('Polls', `/courses/${currentSemester}/sample/polls`);
+    cy.sidebarContains('Course Materials', `/courses/${currentSemester}/sample/course_materials`);
+    cy.sidebarContains('Discussion Forum', `/courses/${currentSemester}/sample/forum`);
+    cy.sidebarContains('My Late Day', `/courses/${currentSemester}/sample/late_table`);
+
+    if (user !== 'student') {
+        cy.sidebarContains('Student Photos', `/courses/${currentSemester}/sample/student_photos`);
+        if (user !== 'ta') {
+            cy.sidebarContains('New Gradeable', `/courses/${currentSemester}/sample/gradeable`);
```

Comment:
Don't you also want to assert that these sidebar items *don't* show up for students?